### PR TITLE
Elisp coding style tweaks

### PR DESCRIPTION
**Pull request by mgrbyte**
@rakanalh This is a bunch of style-related changes that satisfy flycheck's emacs-lisp [checks][1];
Not sure if you want this/use flycheck, but I've done this  anyway whilst creating #14, but separated it out.

[1]: http://www.flycheck.org/en/latest/user/syntax-checkers.html

### CODE REVIEW

**Comment by rakanalh** [comment]
amazing! thanks again.

Would you be so kind squashing the commits here to 1 commit?

File: dashboard.el [hunk 1]
```diff
@@ -254,5 +261,9 @@ If MESSAGEBUF is not nil then MSG is also written in message buffer."
    (redisplay))
   (add-hook 'after-init-hook '(lambda () (switch-to-buffer "*dashboard*"))))
 
+;; Forward declartions for optoinal dependency to keep check-declare happy.
```
Comment (rakanalh) [inline dashboard.el]: typo in optional

**Comment by mgrbyte** [comment]
@rakanalh really sorry, I was evil and forced-pushed the squashed commit which meant I no longer see your review. I have fixed the typo.

File: dashboard.el [hunk 2]
```diff
@@ -254,5 +261,9 @@ If MESSAGEBUF is not nil then MSG is also written in message buffer."
...
+(declare-function projectile-load-known-projects "projectile.el:ext")
+(declare-function projectile-relevant-known-projects "projectile.el:ext")
```
Comment (rakanalh) [inline dashboard.el]: I think it should be the case given this:
https://github.com/jwiegley/emacs-release/blob/master/lisp/emacs-lisp/check-declare.el#L52

**Comment by rakanalh** [comment]
No worries, i guess the last point was about ext being a prefix not a postfix to the package string

**Comment by mgrbyte** [comment]
@rakanalh I've made a bit of a dogs dinner of this PR, will re-issue another one on top the current master.

**Pull request closed without merging**